### PR TITLE
Add more validations for TMA

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

File: csrc/device_lower/validation.cpp [hunk 1]
```diff
@@ -159,13 +159,45 @@ void validateIterDomainUsage(Fusion* fusion) {
 
 void validateCpAsyncBulk(const std::vector<TensorView*>& tvs) {
   for (auto tv : tvs) {
+    bool is_cp_aync_bulk = ir_utils::isCpAsyncBulk(tv->definition());
```
Comment (rdspring1) [inline csrc/device_lower/validation.cpp:162]:
```suggestion
    bool is_cp_async_bulk = ir_utils::isCpAsyncBulk(tv->definition());
```

File: csrc/device_lower/validation.cpp [hunk 2]
```diff
@@ -159,13 +159,45 @@ void validateIterDomainUsage(Fusion* fusion) {
...
+        ParallelType::TIDz,
+        ParallelType::TIDy,
+        ParallelType::TIDx,
+        ParallelType::Unroll,
+        ParallelType::Unswitch,
+        ParallelType::Bulk,
+        ParallelType::Serial};
+    std::unordered_set<IterType> valid_iter_types{
+        IterType::Iteration, IterType::Broadcast};
+    for (auto id : tv->getLoopDomain()) {
```
Comment (rdspring1) [inline csrc/device_lower/validation.cpp:190]: Remove ParallelType::Bulk from `valid_parallel_types'. Then move the `IterType::Iteration` check here.
```cpp
  if (id->getParallelType() == ParallelType::Bulk) {
        NVF_ERROR(
            id->getIterType() == IterType::Iteration,
            "ParallelType::Bulk is only supported for IterType::Iteration.");
        continue;
 }

**Review by rdspring1 — APPROVED** [review]
LGTM.

I made a suggestion that seemed less confusing in my head.

File: csrc/device_lower/validation.cpp (hunk 2 above)
Comment (zasdfgbnm) [reply csrc/device_lower/validation.cpp:189]: changed

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

**Pull request merged**